packages/ui/build/vite/build.mjs

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env zx
+
+import { $ } from 'zx'
+import fs, {
+  existsSync,
+  renameSync,
+  readdirSync,
+  appendFileSync,
+  unlinkSync,
+  statSync,
+  rmSync,
+} from "fs";
+import fsExtra from 'fs-extra'
+const { copySync } = fsExtra
+import readDirRecursive from './utils.mjs'
+
+/**
+ * removing dist dir before we start
+ */
+if (fs.existsSync('./dist')) {
+  fs.rmSync('./dist', { recursive: true })
+}
+
+/**
+ * parallel build for all formats (except ems-node)
+ */
+await Promise.all([
+  $`vite build --config ./build/vite/configs/vite.cjs.js`,
+  $`vite build --config ./build/vite/configs/vite.iife.js`,
+  $`vite build --config ./build/vite/configs/vite.esm.js`,
+  $`vite build --config ./build/vite/configs/vite.styles.js`,
+  $`yarn build:types`,
+])
+
+/**
+ * adding css imports to esm build components (css files generated by config.build.cssCodeSplit)
+ */
+readdirSync('./dist/esm/src/components')
+  .forEach((folderName) => {
+    const currentPath = `./dist/esm/src/components/${folderName}`
+
+    const componentName = folderName.split('-').map((el) => {
+      const split = el.split('')
+      split[0] = split[0].toUpperCase()
+      return split.join('')
+    }).join('')
+
+    const componentFilePath = `${currentPath}/${componentName}.js`
+    const componentCssPath = `${currentPath}/${componentName}.css`
+
+    existsSync(componentFilePath) &&
+    existsSync(componentCssPath) &&
+    appendFileSync(componentFilePath, `\n import './${componentName}.css'`);
+  })
+
+/**
+ * creating esm-node build from esm build copy, renaming files extension from js to mjs
+ */
+copySync('./dist/esm', './dist/esm-node', {})
+readDirRecursive('./dist/esm-node')
+  .filter((el) => el.includes('.js') && !el.includes('.js.map'))
+  .forEach((filePath) => {
+    const newFilePath = filePath.replace('.js', '.mjs')
+    existsSync(filePath) && renameSync(filePath, newFilePath)
```

Comment:
I think you can check `existsSync(filePath)`, and then inside of the `if` assign new variable and call `renameSync` function.